ext/ox/builder.c

```diff
@@ -699,17 +703,21 @@ builder_doctype(VALUE self, VALUE text) {
  *
  * Adds a text element to the XML string being formed.
  * - +text+ - (String) contents of the text field
+ * - +strip_invalid_chars+ - [true|false] strips any characters invalid for XML, defaults to false
  */
 static VALUE
-builder_text(VALUE self, VALUE text) {
+builder_text(int argc, VALUE *argv, VALUE self) {
     Builder		b = (Builder)DATA_PTR(self);
-    volatile VALUE	v = text;
+    volatile VALUE	v;
+    volatile VALUE	strip_invalid_chars;
+
+    rb_scan_args(argc, argv, "11", &v, &strip_invalid_chars);
```

Comment:
This is the hard way with a fair amount of overhead. Just check the value of argc and then check the second arg for the strip option. That assumes this is the approach desired.

Reply:
Sorry, I was reading docs and found `rb_scan_args` useful. I did not think about the overhead, and ox is all about performance. I now check argc.